cmd/branchingconfigmanagers/tide-config-manager/main.go

```diff
@@ -112,133 +111,63 @@ func newSharedDataDelegate() *sharedDataDelegate {
-	openshiftReleaseBranches      sets.String
-	openshiftReleaseBranchesPlus1 sets.String
+type branchingDayEvent struct {
+	repos                    sets.String
+	openshiftReleaseBranches sets.String
 	*sharedDataDelegate
 }
 
-func newFeatureFreezeEvent(current, future string, delegate *sharedDataDelegate) featureFreezeEvent {
-	return featureFreezeEvent{
-		excludedOrgs:                  []string{openshiftPriv},
-		repos:                         sets.NewString(),
-		openshiftReleaseBranches:      sets.NewString(release+current, openshift+current),
-		openshiftReleaseBranchesPlus1: sets.NewString(release+future, openshift+future),
-		sharedDataDelegate:            delegate,
-	}
-}
-
-func (ffe featureFreezeEvent) ModifyQuery(q *prowconfig.TideQuery, repo string) {
-	ffe.ensureFeatureFreezeApprovals(q)
-	if ffe.repos.Has(repo) {
-		ffe.ensureFeatureFreezeBugs(q)
+func newBranchingDayEvent(current, future string, delegate *sharedDataDelegate) branchingDayEvent {
+	return branchingDayEvent{
+		repos:                    sets.NewString(),
+		openshiftReleaseBranches: sets.NewString(release+current, openshift+current),
+		sharedDataDelegate:       delegate,
 	}
 }
 
-func (ffe featureFreezeEvent) GetDataFromProwConfig(pc *prowconfig.ProwConfig) {
-	for _, query := range pc.Tide.Queries {
-		branches := sets.NewString(query.IncludedBranches...)
-		labels := sets.NewString(query.Labels...)
-		ffe.rewriteReposToExcludeOrgs(&query)
-		if branches.Intersection(ffe.openshiftReleaseBranchesPlus1).Len() > 0 && labels.Has(staffEngApproved) {
-			ffe.repos.Insert(query.Repos...)
-			continue
-		}
-		if branches.Intersection(ffe.openshiftReleaseBranches).Len() > 0 && labels.Has(cherryPickApproved) {
-			ffe.repos.Insert(query.Repos...)
-		}
-	}
-}
+func (bde branchingDayEvent) ModifyQuery(q *prowconfig.TideQuery, repo string) {
+	reqLabels := sets.NewString(q.Labels...)
+	branches := sets.NewString(q.IncludedBranches...)
 
-func (ffe *featureFreezeEvent) rewriteReposToExcludeOrgs(q *prowconfig.TideQuery) {
-	repos := []string{}
-	for _, repo := range q.Repos {
-		for _, org := range ffe.excludedOrgs {
-			if strings.Contains(repo, org) {
-				continue
-			}
-			repos = append(repos, repo)
+	if branches.Intersection(bde.openshiftReleaseBranches).Len() > 0 {
+		if reqLabels.Has(staffEngApproved) {
+			reqLabels.Delete(staffEngApproved)
+			reqLabels.Insert(cherryPickApproved, backportRiskAssessed)
 		}
 	}
-	q.Repos = repos
+	q.Labels = reqLabels.List()
 }
 
-func (ffe *featureFreezeEvent) ensureFeatureFreezeBugs(q *prowconfig.TideQuery) {
-	requiredLabels := sets.NewString(q.Labels...)
-	branches := sets.NewString(q.IncludedBranches...)
-	if branches.Intersection(ffe.mainMaster).Len() == 0 {
-		return
-	}
-	if requiredLabels.Intersection(ffe.excludedLabels).Len() > 0 {
-		return
-	}
-	requiredLabels = requiredLabels.Union(ffe.validBug)
-	q.Labels = requiredLabels.List()
+func (bde branchingDayEvent) GetDataFromProwConfig(pc *prowconfig.ProwConfig) {
```

Comment:
It seems that there is no more implementations of `GetDataFromProwConfig`, we could delete this method from `ShardProwConfigFunctors` then. 
Tip: I understand you like [functors](ShardProwConfigFunctors) but it would be nice to rename `ShardProwConfigFunctors` with a name closer to our [business domain](https://en.wikipedia.org/wiki/Domain-driven_design) (branch cuts, in this case) rather than the technical one: it would improve the readability.

Reply:
there is no more implementation in 2 out of 3 implementations. So, we still need that method.